### PR TITLE
feat: specifications for non-EVM payments

### PR DESCRIPTION
**Pull request by yomarion**
## Description of the changes

Publish the specifications from [this document](https://docs.google.com/document/d/1JigWXTe9_P9pw4TjVKcOfBjAjMExkXYtudERcDrXP_k/edit) to our docs.

### CODE REVIEW

File: packages/docs/docs/guides/8-payments/0-intro.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+This is needed ONLY for one of these cases:
+
+- CASE 1: Contract-based currencies such as ERC20 (without custom data)
+- CASE 2: Native coins for blockchains where we cannot send input data with the transaction
+- CASE 3: Native coins with some payments expected to be done by smart wallets
+
+[Cf. the specifications for the Ethreum-based ERC20 proxy with fee](https://github.com/RequestNetwork/requestNetwork/blob/master/packages/advanced-logic/specs/payment-network-erc20-fee-proxy-contract-0.1.0.md)
+
+The code should be strictly equivalent to [the Ethereum version](https://etherscan.io/address/0x370de27fdb7d1ff1e1baa7d11c5820a324cf623c#code).
+You can find several other payment contracts on our Github: https://github.com/RequestNetwork/requestNetwork/tree/master/packages/smart-contracts/test/contracts
```
Comment (alexandre-abrioux) [inline packages/docs/docs/guides/8-payments/0-intro.md:33]: `test` ▶️ `src`
```suggestion
You can find several other payment contracts on our [Github repository](https://github.com/RequestNetwork/requestNetwork/tree/master/packages/smart-contracts/src/contracts).
```

File: packages/docs/docs/guides/8-payments/0-intro.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+[Cf. examples in our repo.](https://github.com/RequestNetwork/requestNetwork/tree/master/packages/payment-detection)
+
+### Index events sent by the proxy
+
+We need a way to query events by paymentReference, as this is the way we can retrieve the payment status of one specific invoice.
+
+### Get notified of events sent by the proxy
+
+What is the preferred way to be notified of new interesting events ?
```
Comment (alexandre-abrioux) [inline packages/docs/docs/guides/8-payments/0-intro.md:74]:
```suggestion
What is the preferred way to be notified of new interesting events? How long does it take for a block to become finalized or to be considered as finalized?
```

**Review by alexandre-abrioux — APPROVED** [review]
Nice! 👍

**Review by benjlevesque — COMMENTED** [review]
discussed in DM: make it clear that this is an advanced usage, not aimed at a developer starting using our tech

**Review by bertux — APPROVED** [review]

**Review by KolevDarko — APPROVED** [review]

**Pull request closed without merging**